### PR TITLE
ICU-22309 update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by srl295**
### Note: this is presumably for v74

-----

- fix broken links
- move all content into CONTRIBUTING.md from https://icu.unicode.org/processes/contribute
- no process change

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22309
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: docs/userguide/dev/contributions.md [hunk 1]
```diff
@@ -32,106 +32,4 @@ upgrade to a new ICU release.
 
 ## Current Process
 
-See <https://icu.unicode.org/processes/contribute>.
-
-## Historical
```
Comment (srl295) [inline docs/userguide/dev/contributions.md:37]: historical section removed

**Comment by markusicu** [comment]
Thanks, Steven, especially for first moving the existing contents basically as-is. Easier to see real changes in a later pull request.

FYI @pedberg-icu @macchiati @richgillam

**Comment by srl295** [comment]
welcome. Am I correct in assuming this needs to wait until v73 is branched?

**Comment by markusicu** [comment]
> Am I correct in assuming this needs to wait until v73 is branched?

Yes... but ICU 73 was branched last Thursday, in preparation for the release candidate. So no need to wait with non-disruptive changes.

**Comment by srl295** [comment]
@markusicu  ready for review and merge then

**Comment by srl295** [comment]
FYI @annebright

**Comment by srl295** [comment]
@markusicu  OK to merge this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,3 +108,25 @@ Please see [./icu4c/LICENSE](./icu4c/LICENSE) (C and J are under an identical li
 Unicode and the Unicode Logo are registered trademarks
 of Unicode, Inc. in the U.S. and other countries.
 [Terms of Use and License](http://www.unicode.org/copyright.html)
+
+
+[coding-guidelines]: docs/userguide/dev/codingguidelines.md
+[git4icu]: https://icu.unicode.org/repository/gitdev
+[unicode-license]: https://www.unicode.org/copyright.html#License
```
Comment (markusicu) [inline CONTRIBUTING.md:115]:
```suggestion
[unicode-license]: https://www.unicode.org/license.txt
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,7 +2,103 @@
...
+We have several [mailing lists][contacts]. Contributors should at least subscribe to the **icu-design** mailing list, and we also have a team-internal list that you should get added to.
+
+We meet once a week by phone. See the [agenda & meeting minutes][meetings].
+
+## Understand ICU
+
+
+*   Home page: <https://icu.unicode.org>
+
+*   User Guide: <https://unicode-org.github.io/icu/userguide/>
```
Comment (markusicu) [inline CONTRIBUTING.md:65]:
```suggestion
*   Home page: https://icu.unicode.org

*   User Guide: https://unicode-org.github.io/icu/userguide/
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,7 +2,103 @@
...
+
+[Tips for developers][tips]
+
+[Eclipse and other setup][setup]
+
+[Submitting ICU Bugs and Feature Requests][bugs]
+
+[ICU Ticket Life cycle][ticket-lifecycle]
+
+Significant/disruptive changes should be discusses on the icu-design list or on the team-internal list.
```
Comment (markusicu) [inline CONTRIBUTING.md:93]:
```suggestion
Significant/disruptive changes should be discussed on the icu-design list or on the team-internal list.
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (srl295) [reply CONTRIBUTING.md:65]: I can do that, just curious about the reason?  Spec:

- [URI Autolink](https://github.github.com/gfm/#uri-autolink) is what I'm using, which is a core part of Markdown
- GFM uses the [`autolink`](https://github.github.com/gfm/#uri-autolink) extension which has more implementation-dependent aspects as to when it is supported.

URIs within `<>` also have a long tradition in plaintext.  `https://…` linkifies in many but not all environments that will read and process the markdown:  think about an IDE previewing this file as part of the user's code browsing experience.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -2,7 +2,103 @@
...
+
+- [git for ICU Developers][git4icu]
+
+[Tips for developers][tips]
+
+[Eclipse and other setup][setup]
+
+[Submitting ICU Bugs and Feature Requests][bugs]
+
+[ICU Ticket Life cycle][ticket-lifecycle]
```
[the remark this answers is not in the document]
Comment (srl295) [reply CONTRIBUTING.md]: better formatting, thanks

**Comment by srl295** [comment]
squashed, took everything except for the angle brackets pending discussion

File: CONTRIBUTING.md (hunk 3 above)
Comment (markusicu) [reply CONTRIBUTING.md:65]: I don't feel super strongly about these, but mostly these days URLs autolink without extra syntax, and the customary URL highlighting has largely replaced the old `<>` highlighting.

If these don't autolink here without the `<>` or if you feel strongly, then keep them as you proposed.

**Review by markusicu — APPROVED** [review]
lgtm tnx

File: CONTRIBUTING.md (hunk 3 above)
Comment (srl295) [reply CONTRIBUTING.md:65]: It works in a lot of places, but it is out of core spec.

Comment (markusicu) [reply CONTRIBUTING.md:65]: Again, I don't care strongly, but I see a lot of Markdown inside Google and elsewhere, and not a lot of `<>`.

**Pull request merged**